sdk/src/org.graalvm.nativeimage/src/org/graalvm/nativeimage/impl/RuntimeSerializationSupport.java

```diff
@@ -41,6 +41,23 @@
 package org.graalvm.nativeimage.impl;
 
 public interface RuntimeSerializationSupport {
+    /**
+     * Register the specified serialization target class itself and all associated classes.
+     * <p>
+     * According to the Java Object Serialization Specification, the associated classes
+     * include 1) all the target class' non-static and non-transient fields types
+     * and their associated classes; 2) other fields defined in the customised
+     * writeObject(ObjectOutputStream) and readObject(ObjectInputStream).
+     * This method can automatically explore all possible serialization target classes
+     * in the first scenario, but can't figure out the classes in the second scenario.
+     * <p>
+     * Another limitation is the specified {@code clazz} must
+     * have no subclasses (effectively final). Otherwise, the actual serialization
+     * target class could be any subclass of the specified class at runtime.
+     *
+     * @param clazz the serialization target class
+     */
+    default void registerAllAssociatedClasses(Class<?> clazz){}
```

Comment:
Why are you providing an empty default implementation?

Reply:
Because for `SerializationDenyRegistry`, it should not register all associated classes as denied. But it should be implemented in the `SerializationDenyRegistry` class. I have fixed this.